spline/components/tasks.py

```diff
@@ -218,6 +219,32 @@ def can_process_shell(self, entry):
 
     def process_shell(self, creator, entry, config):
         """Processing a shell entry."""
+        retries = entry['retries'] if 'retries' in entry else 1
+        counter = 0
+        output = []
+
+        while counter < retries:
```

Comment:
The idea - in general - is ok. How about a retry tool function instead of implementing it specifically in the task module?

Reply:
Not so sure I understand the change request.

Do you mean extract it to its' own function, somewhere? If so, I'd only agree to the change if it is to be reused elsewhere, otherwise it might be an early refactoring change.